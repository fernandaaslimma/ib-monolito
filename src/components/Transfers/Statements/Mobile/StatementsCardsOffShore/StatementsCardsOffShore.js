import React, { Fragment, useState } from "react";
import moment from "moment";
import formatNumber from "../../../../../utils/formatNumber";
import { translate } from "../../../../../utils/i18n";

import {
    CardTitle,
    CardContent,
    Card,
    AvailableAmountHeader,
    AmountsContent,
    AmountsValue,
    Line,
    ItemDescription,
    ItemValue,
    CounterPart,
    StatementInfo
} from "./styles";
import AnimatedBottonSheet from "../../../../common/AnimatedBottomSheet";
import DefaultShimmerLoading from "../../../../common/DefaultShimmerLoading";
import TransferDetails from "../../TransferDetails";

function StatementsCardsOffShore({
    item,
    maskValues,
    formatTitle,
    defaultCurrency,
    loading
}) {
    const [isOpenDetailedBottomSheet, setIsOpenDetailedBottomSheet] = useState(
        false
    );

    const [transferContent, setTransferContent] = useState({
        transferInfo: {},
        receiverInfo: {},
        senderInfo: {},
        transferType: ""
    });
        
    const daysFromToday = moment().diff(item.date, "days");

    const formatValues = value => formatNumber(Math.abs(value), { digits: 2 });

    const triggerModal = transferData => {
        setTransferContent({
            transferInfo: transferData,
            receiverInfo:
            {
                ...transferData.counterparty,
                name:
                    transferData?.counterparty?.name?.length
                        ? transferData.counterparty.name
                        : ""
            }
        });
        setIsOpenDetailedBottomSheet(true);
        return;
    };

    const injectCurrency = value => {
        return `${Math.sign(value) === -1 ? "-" : "+"
            } ${defaultCurrency} ${formatValues(value)}`;
    };

    const renderCounterPartName = (item) => item.counterparty?.name?.length ? item.counterparty.name : "";

    const renderVoucher = () => {
        setIsOpenDetailedBottomSheet(false);
    };

    return loading ? (
        <DefaultShimmerLoading repeat={2} innerRepeat={3} />
    ) : (
        <Fragment>
            <Card data-test="cardOffshore">
                <CardTitle data-test="cardDateOffShore">
                    {item.date && formatTitle(item.date)}
                </CardTitle>
                <CardContent>
                    {item && daysFromToday >= 1 && (
                        <AvailableAmountHeader>
                            <AmountsContent>{translate("DAY_BALANCE")}</AmountsContent>
                            &nbsp;
                            <AmountsValue data-test="amountValueOffShore">
                                {item.finalTotalAmount &&
                                    `${defaultCurrency} ${maskValues(
                                        formatValues(item.finalTotalAmount)
                                    )}`}
                            </AmountsValue>
                        </AvailableAmountHeader>
                    )}

                    {item.transactions &&
                        item.transactions.map((event, i) => {
                            const finalValue = injectCurrency(event.amount);
                            return (
                                <Line
                                    data-test={`cardStatementOffShore${i}`}
                                    isFuture={false}
                                    key={i}
                                    onClick={() => triggerModal(event)}
                                >
                                    <StatementInfo data-test={`cardTransactionTypeOffShore${i}`}>
                                        <ItemDescription value={event.amount}>
                                            {event.description && event.description.toUpperCase()}
                                        </ItemDescription>
                                        <CounterPart>{renderCounterPartName(event)}</CounterPart>
                                    </StatementInfo>
                                    <ItemValue data-test={`cardValueOffShore${i}`} value={event.amount}>
                                        {maskValues(finalValue)}
                                    </ItemValue>
                                </Line>
                            );
                        })}
                </CardContent>
            </Card>
            <AnimatedBottonSheet
                isOpen={
                    isOpenDetailedBottomSheet && Object.keys(transferContent).length > 0
                }
                velocity={0.2}
                head={{ title: transferContent.transferInfo.description }}
                onClickInBack={() => setIsOpenDetailedBottomSheet(false)}
                zIndex={3000}
                dataTest={"BottomSheetTitleOffShore"}
            >
                <TransferDetails
                    transferContent={transferContent}
                    maskValues={maskValues}
                    injectCurrency={injectCurrency}
                    renderVoucher={renderVoucher}
                    closeVoucher={() => setIsOpenDetailedBottomSheet(false)}
                />
            </AnimatedBottonSheet>
        </Fragment>
    );
}

export default StatementsCardsOffShore;
