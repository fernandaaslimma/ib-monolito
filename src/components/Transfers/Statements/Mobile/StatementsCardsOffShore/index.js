import StatementsCardsOffShore from "./StatementsCardsOffShore";

export default StatementsCardsOffShore;
