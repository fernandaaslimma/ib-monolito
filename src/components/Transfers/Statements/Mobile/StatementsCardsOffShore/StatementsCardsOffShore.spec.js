import React from "react";
import StatementsCardsOffShore from "./StatementsCardsOffShore";
import { configure, fireEvent, render } from "@testing-library/react";
import "@testing-library/jest-dom";

configure({ testIdAttribute: "data-test" });

jest.mock("../../../../../utils/i18n", () => ({
    translate: id => id,
    getLanguage: () => "en-US",
    isZhCN: () => false
}));

jest.mock("../../../../../utils/formatDate", () => ({
    getShortDateStringFromEpoch: () => "Wed, 2nd Dec"
}));

const formatTitle = jest.fn(() => "INITIAL DATE");
const maskValues = jest.fn(e => e);

jest.mock("moment", () => () => ({
    diff: (value, parameter) => {
        if (parameter === "years" && value === "2020-12-02T12:33:54.271Z") {
            return 0;
        }
        if (parameter === "days" && value === "2020-12-02T12:33:54.271Z") {
            return 0;
        }
        if (parameter === "days" && value === "2020-12-03T12:33:54.271Z") {
            return 1;
        }
        return 2;
    },
    format: () => "2019"
}));

const props = {
    maskValues,
    defaultCurrency: "USD",
    formatTitle,
    isFuture: false,
    item: {
        date: "2024-07-29",
        finalTotalAmount: 1000000,
        initialTotalAmount: 1200000,
        transactions: [
            {
                amount: 10000,
                description: "TRANSFERENCIA",
                currency: "USD",
                counterparty: {
                    name: "Mattera Prates"
                }
            },
            {
                amount: -1000,
                description: "REGISTRO DE CAPTACAO  EM PRE EXPORT",
                currency: "USD",
                counterparty: {}
            }
        ]
    }
};

describe("Statements Cards OffShore", () => {
    it("should match snapshot", () => {
        expect(render(<StatementsCardsOffShore {...props} />)).toMatchSnapshot();
    });
});

describe("Testing buttons click", () => {
    it("should components initial load", () => {
        const screen = render(<StatementsCardsOffShore {...props} />);

        expect(screen.getByTestId("cardDateOffShore")).toBeInTheDocument();
        expect(screen.getByTestId("amountValueOffShore")).toBeInTheDocument();
        expect(screen.getByTestId("cardStatementOffShore0")).toBeInTheDocument();
        expect(screen.getByTestId("cardTransactionTypeOffShore0")).toBeInTheDocument();
        expect(screen.getByTestId("cardValueOffShore0")).toBeInTheDocument();
        expect(screen.queryByTestId("BottomSheetTitleOffShore")).not.toBeInTheDocument();
    });

    it("should open modal AnimatedBottonSheet", () => {
        const screen = render(<StatementsCardsOffShore {...props} />);

        fireEvent.click(screen.getByTestId("cardStatementOffShore0"));
        expect(screen.getByTestId("BottomSheetTitleOffShore")).toBeInTheDocument();
    });

    it("should open and close modal AnimatedBottonSheet", () => {
        const screen = render(<StatementsCardsOffShore {...props} />);

        // Open Modal
        fireEvent.click(screen.getByTestId("cardStatementOffShore0"));
        expect(screen.getByTestId("BottomSheetTitleOffShore")).toBeInTheDocument();

        //Close Modal
        fireEvent.click(screen.getByTestId("closeVoucherButton"));
        expect(screen.queryByTestId("BottomSheetTitleOffShore")).not.toBeInTheDocument();
    });
});
