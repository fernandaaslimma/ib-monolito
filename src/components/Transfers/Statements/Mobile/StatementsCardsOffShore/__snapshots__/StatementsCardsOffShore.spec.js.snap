// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Statements Cards OffShore should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="overflow: visible;"
  >
    <div>
      <div
        class="sc-EHOje gJLZxB"
        data-test="cardOffshore"
      >
        <p
          class="sc-bdVaJa bsivQC"
          data-test="cardDateOffShore"
        >
          INITIAL DATE
        </p>
        <div
          class="sc-bZQynM foSlkM"
        >
          <div
            class="sc-bxivhb bwhakD"
          >
            <div
              class="sc-bwzfXH eIDKfq"
            >
              DAY_BALANCE
            </div>
             
            <div
              class="sc-htpNat dUqxUp"
              data-test="amountValueOffShore"
            >
              USD 1,000,000.00
            </div>
          </div>
          <li
            class="sc-gzVnrw dUGQdS"
            data-test="cardStatementOffShore0"
          >
            <span
              class="sc-gZMcBi ejcDsQ"
              data-test="cardTransactionTypeOffShore0"
            >
              <div
                class="sc-htoDjs gOrFne"
                value="10000"
              >
                TRANSFERENCIA
              </div>
              <div
                class="sc-dnqmqq jxEiUP"
              >
                Mattera Prates
              </div>
            </span>
            <div
              class="sc-iwsKbI ijhsUm"
              data-test="cardValueOffShore0"
              value="10000"
            >
              + USD 10,000.00
            </div>
          </li>
          <li
            class="sc-gzVnrw dUGQdS"
            data-test="cardStatementOffShore1"
          >
            <span
              class="sc-gZMcBi ejcDsQ"
              data-test="cardTransactionTypeOffShore1"
            >
              <div
                class="sc-htoDjs gOrFne"
                value="-1000"
              >
                REGISTRO DE CAPTACAO  EM PRE EXPORT
              </div>
              <div
                class="sc-dnqmqq jxEiUP"
              />
            </span>
            <div
              class="sc-iwsKbI bfEjBm"
              data-test="cardValueOffShore1"
              value="-1000"
            >
              - USD 1,000.00
            </div>
          </li>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-EHOje gJLZxB"
      data-test="cardOffshore"
    >
      <p
        class="sc-bdVaJa bsivQC"
        data-test="cardDateOffShore"
      >
        INITIAL DATE
      </p>
      <div
        class="sc-bZQynM foSlkM"
      >
        <div
          class="sc-bxivhb bwhakD"
        >
          <div
            class="sc-bwzfXH eIDKfq"
          >
            DAY_BALANCE
          </div>
           
          <div
            class="sc-htpNat dUqxUp"
            data-test="amountValueOffShore"
          >
            USD 1,000,000.00
          </div>
        </div>
        <li
          class="sc-gzVnrw dUGQdS"
          data-test="cardStatementOffShore0"
        >
          <span
            class="sc-gZMcBi ejcDsQ"
            data-test="cardTransactionTypeOffShore0"
          >
            <div
              class="sc-htoDjs gOrFne"
              value="10000"
            >
              TRANSFERENCIA
            </div>
            <div
              class="sc-dnqmqq jxEiUP"
            >
              Mattera Prates
            </div>
          </span>
          <div
            class="sc-iwsKbI ijhsUm"
            data-test="cardValueOffShore0"
            value="10000"
          >
            + USD 10,000.00
          </div>
        </li>
        <li
          class="sc-gzVnrw dUGQdS"
          data-test="cardStatementOffShore1"
        >
          <span
            class="sc-gZMcBi ejcDsQ"
            data-test="cardTransactionTypeOffShore1"
          >
            <div
              class="sc-htoDjs gOrFne"
              value="-1000"
            >
              REGISTRO DE CAPTACAO  EM PRE EXPORT
            </div>
            <div
              class="sc-dnqmqq jxEiUP"
            />
          </span>
          <div
            class="sc-iwsKbI bfEjBm"
            data-test="cardValueOffShore1"
            value="-1000"
          >
            - USD 1,000.00
          </div>
        </li>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
